### DISCUSSION TITLE
[Feature request] Text selection and links through ChatGPT intergration

### DISCUSSION THREAD

**Discussion by Inkapa** [plateaukao/einkbro#259] [Ideas]
Hello,

I've been using the new chatGPT integration on EinkBro, and it's definitely promising. 

In my current use case I've configured the prompt settings to give me a detailed explanation of whatever text I select and I've asked it to always finish by linking sources and additional reading material relating to the matter at hand, as a way to proofread the information it gives me.

However as of now, I can't actually click the links or select any of the text that's displayed in the ChatGPT popup, it'd be a nice improvement to be able to directly open the links ChatGPT gives me in a new tab and to be able to select (and copy) parts of the generated text.

**Comment by plateaukao** [comment]
@Inkapa 
It sounds cool to be able to click on links in chatGPT response. Could you provide me a sample prompt so that I could use to generate linking sources? I'll use it to check if possible to support this feature.

**Reply by Inkapa** [reply]
Of course,

As the ChatGPT role I give it the following prompt: 
> You are a good interpreter. Give detailed explanations about the words, concepts, events, people and sentences I send you. You must always link websites containing sources and additional reading material to research a subject further at the end of your explanation.

And as a prefix I give it this prompt:
> Explain the following while linking your sources:

**Reply by plateaukao** [reply]
@Inkapa 
I tried the prompt in playground of OpenAi website. I found most urls provided are not real.

**Reply by Inkapa** [reply]
It does seem like most urls are indeed fake, I hadn't really realised since I couldn't actually access them. Though I do believe the main idea at hand (enabling links and text selection) is still worth pursuing. 

With the current direction of ChatGPT with GPT4, Bing and the rise of plugins allowing internet access such as https://github.com/interstellard/chatgpt-advanced it might just be a question of waiting a few months before chatGPT can reliably source real webpages.
